crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -202,29 +197,35 @@ impl ModuleSubscriptionActor {
         })
     }
 
-    async fn broadcast_commit_event(&mut self, event: ModuleEvent) -> Result<(), DBError> {
-        async fn _broadcast_commit_event(
-            auth: AuthCtx,
-            mut event: ModuleEvent,
-            subscription: SubscriptionRw,
-            relational_db: Arc<RelationalDB>,
-        ) -> Result<(), DBError> {
-            let ctx = ExecutionContext::incremental_update(relational_db.address());
-            let mut tx = relational_db.begin_tx();
-            let database_update = event.status.database_update().unwrap();
-            let subscription = subscription.read_arc();
-            let futures = FuturesUnordered::new();
+    fn broadcast_commit_event(&mut self, event: ModuleEvent) -> Result<impl Future<Output = ()> + '_, DBError> {
```

Comment:
Nice to see this is no longer async!

Reply:
Yeah, it still is sorta since it returns a future, but it's better encapsulated I feel.